addons/account/models/account_account.py

```diff
@@ -1104,8 +1104,13 @@ def _load_records_write(self, values):
 
     @api.ondelete(at_uninstall=False)
     def _unlink_except_contains_journal_items(self):
-        if self.env['account.move.line'].search_count([('account_id', 'in', self.ids)], limit=1):
-            raise UserError(_('You cannot perform this action on an account that contains journal items.'))
+        for account in self:
+            companies = account.company_ids | self.env.companies
```

Comment:
Why do you need to check the env companies?

Shouldn't you get the account's companies using sudo?